server/document_downloader.py

```diff
@@ -1,64 +1,127 @@
+import datetime
+import lxml.etree as ET
+import re
+import requests
+
 import cachetools.func
 from settings import CACHE_TTL_SECS, CACHE_MAX_SIZE
-import requests
 from flask_restful import abort
 from utils import server_log
 
+
 @cachetools.func.ttl_cache(ttl=CACHE_TTL_SECS, maxsize=CACHE_MAX_SIZE)
 def get_doi_content(user_doi):
     """
     This function is designed to render the paper-journal
-    network for the user given a biorxiv doi
+    network for the user given identifier.
+
+    Code Grabs the latest preprint version to generate results.
+
     Args:
-        user_doi - a biorxiv doi that grabs the most current version of a preprint
+        user_doi - user provided biorxiv/medrxiv doi or arxiv id
     """
 
-    # Try pinging biorxiv server first
-    content = ping_biorxiv_or_medrxiv(user_doi, server="biorxiv")
-    doc_url = f"http://biorxiv.org/content"
+    # create flag if xml is found
+    xml_found = False
+
+    # Check to see if user_doi is a biorxiv/medrxiv doi
+    # if not then resort to searching for arxiv
+    bio_med_rxiv = re.search(r"10\.1101\/", user_doi) is not None
 
-    # If no match found try medrxiv
-    if content is None:
-        content = ping_biorxiv_or_medrxiv(user_doi, server="medrxiv")
-        doc_url = f"http://medrxiv.org/content"
+    if bio_med_rxiv:
+        # Try pinging biorxiv server first
+        content = ping_biorxiv_or_medrxiv(user_doi, server="biorxiv")
+        doc_url = f"http://biorxiv.org/content"
 
-        # If no match at all then raise the red flag
+        # If no match found try medrxiv
         if content is None:
```

Comment:
Question on how flask works: shouldn't `abort` within `ping_biorxiv_or_medrxiv` raise an exception? Why does the code ever get to the point where it can return None?

Reply:
>Question on how flask works: shouldn't abort within ping_biorxiv_or_medrxiv raise an exception? Why does the code ever get to the point where it can return None?

Good question. The `abort`function is for the event that our server doesn't receive any response back from biorxiv's api server. If `abort` executes, you are right as none of the other code will execute. Majority of requests we make to biorxiv gives back a response; however, the response may not return anything useful (e.g., gives back an empty collection of documents)